GTG/gtk/general_preferences.py

```diff
@@ -0,0 +1,252 @@
+from GTG.core.translations import _
+from GTG.tools.autostart import Autostart
+from GTG.tools import shortcut
+
+AUTOSTART_DIRECTORY = os.path.join(xdg_config_home, "autostart")
+AUTOSTART_FILE = "gtg.desktop"
+AUTOSTART_PATH = os.path.join(AUTOSTART_DIRECTORY, AUTOSTART_FILE)
+
+
+class GeneralPreferences(object):
+
+    GENERAL_PREFERENCES_UI = os.path.join(UI_DIR, "general_preferences.ui")
+    INVALID_COLOR = Gdk.Color(50000, 0, 0)
+
+    def __init__(self, req, vmanager):
+        self.req = req
+        self.config = self.req.get_config('browser')
+        builder = Gtk.Builder()
+        builder.add_from_file(self.GENERAL_PREFERENCES_UI)
+
+        self.window = builder.get_object("general_pref_window")
+        self.autostart_button = builder.get_object("autostart_button")
+        self.preview_button = builder.get_object("preview_button")
+        self.bg_color_button = builder.get_object("bg_color_button")
+        self.shortcut_button = builder.get_object("shortcut_button")
+        self.font_button = builder.get_object("font_button")
+        self.shortcut_popover = builder.get_object("shortcut_popover")
+        self.set_shortcut = builder.get_object("set_shortcut")
+
+        self.shortcut = ShortcutWidget(builder)
+
+        self.timer = vmanager.timer
+        self.refresh_time = builder.get_object("time_entry")
+        editor_font = self.config.get("font_name")
+        if editor_font == "":
+            font = self.window.get_style_context().get_font(
+                Gtk.StateFlags.NORMAL)
+            editor_font = font.to_string()
+        self.font_button.set_font_name(editor_font)
+
+        builder.connect_signals(self)
+
+    def get_ui(self):
+        return self.window
+
+    def on_shortcut_popover(self, widget):
+        self.shortcut_popover.show()
+
+    def _refresh_preferences_store(self):
+        """ Sets the correct value in the preferences checkboxes """
+        has_autostart = os.path.isfile(AUTOSTART_PATH)
+        self.autostart_button.set_active(has_autostart)
+
+        self.shortcut.refresh_accel()
+
+        show_preview = self.config.get("contents_preview_enable")
+        self.preview_button.set_active(show_preview)
+
+        bg_color = self.config.get("bg_color_enable")
+        self.bg_color_button.set_active(bg_color)
+        formated_time = self.timer.get_configuration().strftime("%X")
+        correct_locale_time = re.sub(r'([0-9]{1,2}:[0-9]{2}):[0-9]{2}', r'\1',
+                                     formated_time)
+        self.refresh_time.set_text(correct_locale_time)
+
+    def _refresh_task_browser(self):
+        """ Refresh tasks in task browser """
+        task_tree = self.req.get_tasks_tree(refresh=False).get_basetree()
+        task_tree.refresh_all()
+
+    def on_valid_check(self, widget):
+        try:
+            input_time = self.refresh_time.get_text()
+            self.timer.parse_time(input_time)
+            color = None
+
+        except (ValueError, TypeError):
```

Comment:
hmm, good point, I think there was a reason for having it in previous implementations of my parse_time(), however not anymore :) thanks!